### PR TITLE
Allow validate_subject_confirmation Response validation to be skipped

### PR DESCRIPTION
**Pull request by sthanson**
## Status
**READY**

## Migrations
NO

## Description
When you are initializing a new OneLogin::RubySaml::Response, this adds an additional option called skip_subject_confirmation.  When skip_subject_confirmation is true, the validate_subject_confirmation validation is skipped

## Related PRs
List related PRs against other branches:

none

## Todos
- [x] Tests
- [x] Documentation

## Deploy Notes
No deploy notes.  After the deploy there will be a new option available to you.

## Steps to Test or Reproduce
Run validations on an assertion with a bad Subject > SubjectConfirmation.  They should fail.
Then initialize a OneLogin::RubySaml::Response with {skip_subject_confirmation: true} in the options.  Now the bad assertion will pass validations.

## Impacted Areas in Application
This affects the saml validations that are run on your Response.  By default nothing will change.  If you pass {skip_subject_confirmation: true} to the OneLogin::RubySaml::Response, the subject confirmation validation will be skipped

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,6 +166,13 @@ def saml_settings
 end
 ```
 
+Some assertion validations may be skipped by passing parameters to OneLogin::RubySaml::Response.new().  For example, you can skip the Conditions validation or the SubjectConfirmation validations by initializing the response with different options:
```
Comment (pitbulk) [inline README.md]: Replace
```
Some assertion validations may be
```
by
```
Some assertion validations can be
```

**Comment by pitbulk** [comment]
I made a comment and if you can re-build the PR great, if not I can still merge it.

**Comment by sthanson** [comment]
Closing this pull request in favor of the cleaner one without a merge commit:
https://github.com/onelogin/ruby-saml/pull/261

**Comment by levelboy** [comment]
@sthanson I like the description you put on your PRs. Do you write them by hand or do you generate them with a tool?

**Pull request closed without merging**